### PR TITLE
PS-7267: [Reworked] Implement CI FS tests

### PR DESCRIPTION
**Pull request by Sudokamikaze**
Builds: https://ps80.cd.percona.com/job/percona-server-8.0-param-ci/9/

### CODE REVIEW

File: jenkins/pipeline.groovy [hunk 1]
```diff
@@ -240,6 +244,15 @@ pipeline {
                                     sleep 5
                                 done
 
+                                if [[ \$CI_FS_MTR == 'yes' ]]; then
+                                    if [[ ! -f /mnt/ci_disk_\$CMAKE_BUILD_TYPE.img ]] && [[ -z \$(mount | grep /mnt/ci_disk_dir_\$CMAKE_BUILD_TYPE) ]]; then
+                                        sudo dd if=/dev/zero of=/mnt/ci_disk_\$CMAKE_BUILD_TYPE.img bs=1G count=10
+                                        sudo /sbin/mkfs.vfat /mnt/ci_disk_\$CMAKE_BUILD_TYPE.img
+                                        sudo mkdir -p /mnt/ci_disk_dir_\$CMAKE_BUILD_TYPE
+                                        sudo mount -o loop -o uid=27 -o gid=27 -o check=r /mnt/ci_disk_\$CMAKE_BUILD_TYPE.img /mnt/ci_disk_dir_\$CMAKE_BUILD_TYPE
```
Comment (percona-ysorokin) [inline jenkins/pipeline.groovy:252]: Magic numbers `27`
May be
```
-o uid=`id -u ...` -o gid=`id - g ...`
```

File: local/test-binary [hunk 2]
```diff
@@ -195,4 +195,22 @@ if [[ "${TOKUDB_ENGINES_MTR}" = 'no' ]] && [[ "${HOTBACKUP_TESTING}" = 'no' ]] &
...
+        CI_TESTS+=" $CI_TESTS_TMP"
+    done
+
+    MTR_BUILD_THREAD=auto ./mtr \
+        --force --mysqld-env="LD_PRELOAD=${MYSQLD_ENV}" \
+        --max-test-fail=0 \
+        --parallel=$(nproc) \
+        --retry=0 \
+        --tmpdir="/tmp/ps_mtr_tempdir" \
+        --vardir="/tmp/ci_disk_dir" \
```
Comment (percona-ysorokin) [inline local/test-binary:211]: Also add `--big-test`

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

File: jenkins/pipeline.groovy (hunk 1 above)
Comment (Sudokamikaze) [reply jenkins/pipeline.groovy:252]: Those IDs are from docker image:
```
uid=27(mysql) gid=27(mysql) groups=27(mysql)
```
If we use `id -u` it'll return id of the worker user

Comment (Sudokamikaze) [reply jenkins/pipeline.groovy:252]: And this ID/GID is hardcoded here: https://github.com/Percona-Lab/ps-build/blob/8.0/docker/Dockerfile.inc#L9

Comment (Sudokamikaze) [reply jenkins/pipeline.groovy:252]: To be exact, it'll return:
```
uid=1000(ec2-user) gid=1000(ec2-user)
```

Comment (percona-ysorokin) [reply jenkins/pipeline.groovy:252]: Actually, I meant
```
-o uid=`id -u mysql` -o gid=`id - g mysql`
```
But I think I understand that it's not possible as there is no `mysql` user outside docker image (in the worker OS).
So, hardcoded values are fine then.

File: local/test-binary (hunk 2 above)
Comment (Sudokamikaze) [reply local/test-binary:211]: Added in 3f0b1cc
Builds: https://ps80.cd.percona.com/job/percona-server-8.0-param-ci/10/

File: jenkins/pipeline.groovy (hunk 1 above)
Comment (Sudokamikaze) [reply jenkins/pipeline.groovy:252]: It's not possible, as this user don't exist on worker node, only inside ps-build image

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Review by EvgeniyPatlan — APPROVED** [review]

**Pull request merged**